xrootd/xrdfuse/xrdfuse.go

```diff
@@ -74,8 +76,20 @@ func (fs *FS) GetAttr(name string, ctx *fuse.Context) (*fuse.Attr, fuse.Status)
 // OpenDir implements pathfs.FileSystem.OpenDir
 func (fs *FS) OpenDir(name string, ctx *fuse.Context) (c []fuse.DirEntry, code fuse.Status) {
 	entries, err := fs.xrdfs.Dirlist(context.Background(), path.Join(fs.root, name))
+	if serverError, ok := err.(xrdproto.ServerError); ok {
+		switch serverError.Code {
```

Comment:
Perhaps all this handling (and in other functions) could be factorized in a function that translates an xrd error into a fuse one?